### PR TITLE
chore: update makefile structure and functionality

### PR DESCRIPTION
**Pull request by danielgines**
Improvements:
- Reorganized commands to facilitate understanding of the structure
- Enhanced check_tools to detect the Python version
- Added a multi-environment script to support check_tools
- make init now builds the frontend and runs the application
- Aesthetic improvements in output messages

TO-DO:
- Reorganize container-related commands
- Reorganize other miscellaneous utilities
- Document usage in the application docs
- Prepare the dev environment following the maintainers' recommended practices

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,33 +21,79 @@ workers ?= 1
...
+	@echo '----'
+
+######################
+# INSTALL PROJECT
+######################
+
+install_backend: ## install the backend dependencies
+	@echo 'Installing backend dependencies'
+	@poetry install
+	@poetry run pre-commit install
```
Comment (ogabrielluiz) [inline Makefile:59]: We are not using pre-commit anymore.

File: Makefile [hunk 2]
```diff
@@ -338,24 +396,3 @@ endif
 ifdef main
 	make publish_langflow
 endif
-
-patch: ## bump the version in langflow and langflow-base
```
Comment (ogabrielluiz) [inline Makefile:342]: I think we should keep this.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @danielgines 

This looks really good. Just got some comments and we can merge this.

File: Makefile (hunk 2 above)
Comment (danielgines) [reply Makefile:342]: Of course, my mistake, I restored the patch to the Makefile.

File: Makefile (hunk 1 above)
Comment (danielgines) [reply Makefile]: I removed the pre-commit.

**Review by ogabrielluiz — APPROVED** [review]
LGTM

**Pull request merged**